src/charm.py

```diff
@@ -227,6 +224,8 @@ def _on_peer_relation_departed(self, event: RelationDepartedEvent) -> None:
             event.defer()
             return
 
+        self._patroni.update_synchronous_node_count(len(self._units_ips))
```

Comment:
`UpdateSyncNodeCountErrorr` (or the `RetryError`) is not being handled here and in the other call.